packages/context/src/__tests__/acceptance/context-view.acceptance.ts

```diff
@@ -0,0 +1,184 @@
+import {expect} from '@loopback/testlab';
+import {
+  Binding,
+  filterByTag,
+  Context,
+  ContextObserver,
+  ContextEventType,
+  ContextView,
+  Getter,
+  inject,
+} from '../..';
+
+let app: Context;
+let server: Context;
+
+describe('ContextView', () => {
+  let contextView: ContextView;
+  beforeEach(givenViewForControllers);
+
+  it('watches matching bindings', async () => {
+    // We have server: 1, app: 2
+    expect(await getControllers()).to.eql(['1', '2']);
+    server.unbind('controllers.1');
+    // Now we have app: 2
+    expect(await getControllers()).to.eql(['2']);
+    app.unbind('controllers.2');
+    // All controllers are gone from the context chain
+    expect(await getControllers()).to.eql([]);
+    // Add a new controller - server: 3
+    givenController(server, '3');
+    expect(await getControllers()).to.eql(['3']);
+  });
+
+  function givenViewForControllers() {
+    givenServerWithinAnApp();
+    contextView = server.createView(filterByTag('controller'));
+    givenController(server, '1');
+    givenController(app, '2');
+  }
+
+  function givenController(_ctx: Context, _name: string) {
+    class MyController {
+      name = _name;
+    }
+    _ctx
+      .bind(`controllers.${_name}`)
+      .toClass(MyController)
+      .tag('controller');
+  }
+
+  async function getControllers() {
+    // tslint:disable-next-line:no-any
+    return (await contextView.values()).map((v: any) => v.name);
+  }
+});
+
+describe('@inject.* - injects a live collection of matching bindings', async () => {
+  beforeEach(givenPrimeNumbers);
+
+  class MyControllerWithGetter {
+    @inject.getter(filterByTag('prime'))
+    getter: Getter<number[]>;
+  }
+
+  class MyControllerWithValues {
+    constructor(
+      @inject(filterByTag('prime'))
+      public values: number[],
+    ) {}
+  }
+
+  class MyControllerWithView {
+    @inject.view(filterByTag('prime'))
```

Comment:
All these `@inject.*` that takes a binding filter function is backed by context view. If we want to move, we probably should have a `dependency-injection.acceptance` test. Let's leave it out of this PR.

Reply:
> All these `@inject.*` that takes a binding filter function is backed by context view. If we want to move, we probably should have a `dependency-injection.acceptance` test. Let's leave it out of this PR.

Acceptance tests should be organized around features and use-cases, not implementation details. It's important to make it easy for contributors (including us!) to find the right place where to add a new test. For example, when there is a bug in `@inject.getter`, then how is the developer supposed to know where to put the new test?

I am fine with leaving reorganization of existing test code out of scope of this pull request. That does not apply to new tests added by your pull request. Let's define the new test structure now, so that your new tests can start following the right way and we avoid code churn.